provisioning/service/src/Contract/ContractApiHttp.cs

```diff
@@ -46,25 +46,32 @@ internal class ContractApiHttp : IContractApiHttp
         public ContractApiHttp(
             Uri baseAddress,
             IAuthorizationHeaderProvider authenticationHeaderProvider,
-            ProvisioningServiceHttpSettings httpSettings)
+            ProvisioningServiceClientOptions options)
         {
             _baseAddress = baseAddress;
             _authenticationHeaderProvider = authenticationHeaderProvider;
 
+            if (options.HttpClient != null)
```

Comment:
Are you sure `options` can't be null here?

Reply:
Yes. We deliberately ensure that options is never null even if the service client is constructed like

```csharp
new ProvisioningServiceClient("connString", null);
```